### PR TITLE
Bug/coinbase auth

### PR DESCRIPTION
**Pull request by reddavis**
## Branch

`master`

## Overall Issue 

I couldn't add a Coinbase account

### Issue 1

After authorizing a Coinbase account the Balance app would not reopen.

#### Steps To Reproduce

- Open menu bar app
- Click settings button
- Click "Add Account"
- Select "Coinbase"
- On Coinbase website, click "Authorize"
- App doesn't open

#### Reproduce Rate

Everytime

#### Fix

I added the `balancemymoney` url scheme back.

### Issue 2

After fixing issue 1, I noticed errors being risen about incorrect JSON when extracting auth data. (`CoinbaseApi.swift#82`)

#### Steps To Reproduce

- Open menu bar app
- Click settings button
- Click "Add Account"
- Select "Coinbase"
- On Coinbase website, click "Authorize"
- App open
- Auth fails

#### Reproduce Rate

Everytime

#### Fix

I updated the JSON parsing

### Issue 3

After fixing issue 2, I then noticed that my accounts weren't appearing. The error looked to be JSON parsing, though when I looked at the API documentation it looked correct.

I was then weirded out at how the API was returning different JSON to what the docs said. Looking closer I noticed Coinbase reported this warning:

```
    warnings =     (
                {
            id = "invalid_version";
            message = "Flux Capacitor Overheated: API version can't be in the future";
            url = "https://developers.coinbase.com/api#versioning";
        }
    );
```

#### Steps To Reproduce

- Open menu bar app
- Click settings button
- Click "Add Account"
- Select "Coinbase"
- On Coinbase website, click "Authorize"
- App open
- Auth succeeds
- No accounts are added

#### Reproduce Rate

Everytime

#### Fix

I changed the `CB-VERSION` to the latest version.

### CODE REVIEW

File: Balance/Shared/Data Model/APIs/Coinbase/CoinbaseApi.swift [hunk 1]
```diff
@@ -76,9 +76,12 @@ struct CoinbaseApi {
                 guard let data = maybeData, maybeError == nil else {
                     throw BalanceError.noData
                 }
-
+                
                 // Try to parse the JSON
-                guard let JSONResult = try JSONSerialization.jsonObject(with: data, options: .mutableContainers) as? [String: AnyObject], let accessToken = JSONResult["accessToken"] as? String, accessToken.length > 0, let refreshToken = JSONResult["refreshToken"] as? String, refreshToken.length > 0, let expiresIn = JSONResult["expiresIn"] as? TimeInterval else {
+                guard let JSONResult = try JSONSerialization.jsonObject(with: data, options: .mutableContainers) as? [String: AnyObject],
+                      let accessToken = JSONResult["access_token"] as? String, accessToken.length > 0,
```
Comment (wolffan) [inline Balance/Shared/Data Model/APIs/Coinbase/CoinbaseApi.swift]: they have changed the way they send JSON? omg

Comment (einsteinx2) [reply Balance/Shared/Data Model/APIs/Coinbase/CoinbaseApi.swift]: This comes from our own server, which should not have changed. So something is not right here. I'm going to investigate this further before merging this pr.

**Comment by einsteinx2** [comment]
Ahh I see what happened. The app was pointing to the new backend since the coinbase errors PR I think and I must have broken something in the new backend during a refactor and started passing the coinbase response directly instead of repackaging it (all the more reason for more tests!). Fixing the backend now so most of this change won't be necessary. 

The info.plist getting clobbered is a bigger question... Git says it was changed on Aug 25, but I swear I was able to add a Coinbase account since then. I must have been in a different branch. All the more reason for more teeeeeeests!! :)

**Comment by einsteinx2** [comment]
Closing this PR, fixing the backend response issue, and restoring the info.plist from the previous good version in another branch.

**Pull request closed without merging**